scripts/priceBoundPSMDeployUSDC.ts

```diff
@@ -0,0 +1,167 @@
+import hre, { ethers } from 'hardhat';
+import { expect } from 'chai';
+import config from './Config';
+import { getImpersonatedSigner } from '@test/helpers';
+
+/// ---------------------------------------------------------------- ///
+///                                                                  ///
+///                Price Bound Peg Stability Module                  ///
+///                                                                  ///
+/// ---------------------------------------------------------------- ///
+
+/// Description: This module is used to manage the stability of the peg.
+
+/// Steps:
+///  0 - Deploy USDC/VOLT PriceBoundPegStabilityModule
+///  1 - Pause Redemptions on the VOLT PriceBoundPegStabilityModule
+///  2 - Move VOLT into the PriceBoundPegStabilityModule to allow trades
+
+/// This contract needs no roles in the system, just funding to allow swaps
+
+/// ---------------------------------------------------------------- ///
+
+const {
+  CORE,
+  USDC,
+  VOLT,
+  PRICE_BOUND_PSM,
+  VOLT_FUSE_PCV_DEPOSIT,
+  MINT_FEE_BASIS_POINTS,
+  REDEEM_FEE_BASIS_POINTS,
+  PROTOCOL_MULTISIG_ADDRESS,
+  ORACLE_PASS_THROUGH_ADDRESS
+} = config;
+
+const usdcReservesThreshold = ethers.utils.parseEther('10000000000'); /// this is greater than the trillions because USDC only has 6 decimals
+const mintLimitPerSecond = ethers.utils.parseEther('10000');
+const voltPSMBufferCap = ethers.utils.parseEther('10000000');
+const initialUSDCPSMVoltAmount = ethers.utils.parseEther('1941000');
+
+/// Oracle price gets scaled up by 1e12 to account for the differences in decimals of USDC and VOLT.
+/// USDC has 6 decimals while Volt has 12, thus creating a difference that has to be normalized
+const voltUSDCDecimalsNormalizer = 12;
+/// Need to scale up price of floor and ceiling by 1e12 to account for decimal normalizer that is factored into price
+const voltFloorPrice = '9000000000000000';
+const voltCeilingPrice = '10000000000000000';
+
+/// Deploy the USDC PriceBoundPSM
+const deploy = async () => {
+  const voltPSMFactory = await ethers.getContractFactory('PriceBoundPSM');
+
+  /// Deploy USDC Peg Stability Module
+  /// PSM will trade VOLT between 100 cents and 111 cents. If the Volt price exceeds 111 cents,
```

Comment:
Not sure I follow this. This PSM is being used as a way to accumulate USDC right. Redemptions are paused, meaning VOLT can only be minted in exchange for providing USDC. 

Why is the VOLT price range set to $1 - $1.11 and not $0.89 - $1 in that case? Setting to $1 - $1.11 makes VOLT expensive relative to USDC, meaning users will not want to provide USDC underlying? The floor and ceiling prices won't be inverted

Reply:
the oracle price is reported inverted, so you have to invert both floor and ceiling prices to get expected behavior